src/cmd/serve.rs

```diff
@@ -92,16 +92,34 @@ fn set_serve_error(msg: &'static str, e: errors::Error) {
     }
 }
 
-async fn handle_request(req: Request<Body>, mut root: PathBuf) -> Result<Response<Body>> {
+async fn handle_request(
```

Comment:
This is starting to have a lot of logic that is not tested :/ 
I don't know the story with hyper in terms of tests, is it easy to add some to test this endpoint?

Reply:
I definitely agree there should be a more thorough testing framework around this (these changes and the preexisting logic). I would consider that outside the scope of this PR though as there was no endpoint testing on `serve.rs` prior and these particular changes are more of an issue fix than an enhancement.

I would be happy to open a subsequent PR to enhance the testing around this.

Reply:
Yep that sounds good, thanks!